api/helpers.go

```diff
@@ -4,42 +4,34 @@ import (
 	"encoding/hex"
 	"encoding/json"
 	"errors"
-	"fmt" // required for evm encoding
+	"fmt"
 	"math/big"
 	"reflect"
-	"strings"
 
 	cmtpool "github.com/cometbft/cometbft/mempool"
 	ctypes "github.com/cometbft/cometbft/rpc/core/types"
 	"github.com/ethereum/go-ethereum/accounts/abi"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/iancoleman/strcase"
 	"go.vocdoni.io/dvote/types"
+	"go.vocdoni.io/dvote/vochain/indexer/indexertypes"
 	"go.vocdoni.io/proto/build/go/models"
 	"google.golang.org/protobuf/encoding/protojson"
 	"google.golang.org/protobuf/proto"
 )
 
-func (a *API) electionSummaryList(pids ...[]byte) ([]*ElectionSummary, error) {
-	processes := []*ElectionSummary{}
-	for _, pid := range pids {
-		// TODO(mvdan): We construct ElectionSummaries in many places, deduplicate.
-		// TODO(mvdan): ProcessInfo could give us the results envelope height as well.
-		procInfo, err := a.indexer.ProcessInfo(pid)
-		if err != nil {
-			return nil, ErrCantFetchElection.WithErr(err)
-		}
-		processes = append(processes, &ElectionSummary{
-			ElectionID:     procInfo.ID,
-			OrganizationID: procInfo.EntityID,
-			Status:         strings.ToLower(models.ProcessStatus_name[procInfo.Status]),
-			StartDate:      procInfo.CreationTime,
-			EndDate:        a.vocinfo.HeightTime(int64(procInfo.EndBlock)),
-			FinalResults:   procInfo.FinalResults,
-			VoteCount:      procInfo.VoteCount,
-		})
+func (a *API) electionSummary(pi *indexertypes.Process) ElectionSummary {
+	return ElectionSummary{
+		ElectionID:     pi.ID,
+		OrganizationID: pi.EntityID,
+		// XXX: some callers used strings.ToLower on Status; what should we consolidate on?
+		Status: models.ProcessStatus_name[pi.Status],
```

Comment:
i checked this and it's a tough decision. ideally i'd love to have consistency between internal code and what's exposed in the API, but it doesn't seem possible. using lowercase in our code looks like a no-go. 

and uppercase in API would look funny in `/accounts/{organizationID}/elections/status/{status}/page/{page}` or in the reply to that query.

so i vote for strings.ToLower , so that API replies are lowercase and all of our internal code is kept uppercase

our public docs are not uppercase
https://developer.vocdoni.io/protocol/overview#election-lifecycle-states

Reply:
@p4u said:

> I think we should consolidate it to the name defined in the protobuf file. So it if the enum is using capital letters, we should return capital letters. The consumers of the API use the same protobuf file.

I tend to think the same - unless there's a really good reason to avoid uppercase, we should use the same casing everywhere, including protobuf.

Unless the existing clients already assume lowercase in lots of places and changing that could be problematic. Going all consistent on lowercase would also be fine by me as long as everyone is happy on the client side.

A third option is to always return the canonical form (uppercase, like protobuf), but queries could accept either uppercase or lowercase forms. This is probably what breaks the fewest clients right now, but it's also the trickiest to deal with in the long term, because it could lead to subtle bugs.

Reply:
SGTM :v:

CAPS FTW

(everywhere, and fix the few places that might be using lowercase ATM)

Reply:
Just removing the XXX comment then, since we want to return the uppercase protobuf casing directly :)